GitUI/Infrastructure/PuttyHelpers.cs

```diff
@@ -68,7 +68,10 @@ public static bool StartPageantIfConfigured(Func<string?> sshKeyFileLoader)
             ThrowIfFileNotFound(AppSettings.Pageant, $"'{AppSettings.Pageant}'\r\n\r\n{TranslatedStrings.ErrorSshPuTTYInstalled}");
 
             string? sshKeyFile = sshKeyFileLoader();
-            ThrowIfFileNotFound(sshKeyFile, $"'{sshKeyFile}'", TranslatedStrings.ErrorSshKeyNotFound);
+            if (!File.Exists(sshKeyFile))
+            {
+                return false;
```

Comment:
Dummy question: Why not start Pageant anyway and bail out before `RunCommand(keyfile)`?

Reply:
Could do even better. Check if already running and then start only if not running.

Reply:
Yes, switch the order.
Has anyone tested the setup?

Reply:
https://github.com/gitextensions/gitextensions/blob/6eab7392839c4d103bad1581fba5eaf6f008d766/GitUI/CommandsDialogs/FormPull.cs#L893-L916

If I'm not mistaken, this was the original logic - collect keys and, if those exist, start PuTTY.

I don't use SSH or keys, so I'm not setup to test it.

Reply:
> I don't use SSH or keys, so I'm not setup to test it.

Me too. That's why "dummy question".

Reply:
https://www.howtogeek.com/devops/should-you-use-https-or-ssh-for-git/#:~:text=Where%20SSH%20takes,over%20the%20wire.  Why I use it.

Reply:
Thank you for the guide.
Plink seems to be hanging on me while trying to connect:
https://github.com/gitextensions/gitextensions/blob/21345f29511f71fb1ca7cfd8ea128972b56863e4/GitUI/Infrastructure/Plink.cs#L74-L82

![image](https://user-images.githubusercontent.com/4403806/208422753-bade5b5f-da62-4001-b0eb-51cbb5b1f30e.png)

Reply:
I didn't have PuTTY installed, and I installed the latest one available: https://the.earth.li/~sgtatham/putty/latest/w64/putty-64bit-0.78-installer.msi

Reply:
Omg.... I had to type "y" in this window so that a new fingerprint was recorded. Then I had to cancel the whole sequence and go restart the putty agent... t😩

Reply:
Oh yeah. That's the putty am I talking to valid server. Only have to do that once.per server.

Reply:
start plink should be before the key check
Otherwise this PR is approved

Reply:
Do you mean like this?
```diff
diff --git a/GitUI/Infrastructure/PuttyHelpers.cs b/GitUI/Infrastructure/PuttyHelpers.cs
index b2b8ddff0..fb07fe2ea 100644
--- a/GitUI/Infrastructure/PuttyHelpers.cs
+++ b/GitUI/Infrastructure/PuttyHelpers.cs
@@ -64,29 +64,29 @@ public static bool StartPageantIfConfigured(Func<string?> sshKeyFileLoader)
                 throw new UserExternalOperationException(TranslatedStrings.ErrorSshPuTTYWhereConfigure,
                     new ExternalOperationException(innerException: new Exception(TranslatedStrings.ErrorSshPuTTYNotConfigured)));
             }
 
             ThrowIfFileNotFound(AppSettings.Pageant, $"'{AppSettings.Pageant}'\r\n\r\n{TranslatedStrings.ErrorSshPuTTYInstalled}");
 
-            string? sshKeyFile = sshKeyFileLoader();
-            if (!File.Exists(sshKeyFile))
-            {
-                return false;
-            }
-
             Executable pageantExecutable = new(AppSettings.Pageant);
 
             // ensure pageant is loaded, so we can wait for loading a key in the next command
             // otherwise we'll stuck there waiting until pageant exits
             if (!IsPageantRunning())
             {
                 // NOTE we leave the process to dangle here
                 IProcess process = pageantExecutable.Start();
                 process.WaitForInputIdle();
             }
 
+            string? sshKeyFile = sshKeyFileLoader();
+            if (!File.Exists(sshKeyFile))
+            {
+                return false;
+            }
+
             return pageantExecutable.RunCommand(sshKeyFile.Quote());
 
             static bool IsPageantRunning()
             {
                 var pageantProcName = Path.GetFileNameWithoutExtension(AppSettings.Pageant);
                 return Process.GetProcessesByName(pageantProcName).Length != 0;

```

Could you please help me understand why we'd want to start plink if we don't have a key?

Reply:
From the feedback of the other issues we basically want to have 2 cases. 

If key file is blank then still attempt connect can suggest key file be set but not required in some alert. 

If key file not blank and file doesn't exist then show error that can't find key file.

Reply:
See https://github.com/gitextensions/gitextensions/issues/10530#issuecomment-1354346577

Reply:
@vbjay I still having troubles understanding why this is necessary. In the original implementation we would only start putty is we had a valid key.
https://github.com/gitextensions/gitextensions/blob/6eab7392839c4d103bad1581fba5eaf6f008d766/GitUI/CommandsDialogs/FormPull.cs#L893-L916
https://github.com/gitextensions/gitextensions/blob/6eab7392839c4d103bad1581fba5eaf6f008d766/GitCommands/Git/GitModule.cs#L1429-L1439

@SymbioticKilla confirmed in https://github.com/gitextensions/gitextensions/issues/10530#issuecomment-1357709486 that the current change is working as expected.

Reply:
So in putty you can save sessions that are configured ri have all their config setup. So when its used this way the server address used ditects it to use x key. 

Yes. I still like putting the key in my config because I can run grep or whatever to find all repos that use x key by examining the config files. But others are managing in putty so we need to support both flows. 

1. Doing it the way we do it now with key files
1. Leaving key file blank and still able to connect because we trust putty to load it up. But we still check result and I'd failed can ask did you include key file or whatnot.